x-pack/plugins/secops/server/utils/serialized_query.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { UserInputError } from 'apollo-server-errors';
+
+import { JsonObject } from '../../common/typed_json';
+
+export const parseFilterQuery = (
+  filterQuery: string | null | undefined
+): JsonObject | undefined => {
+  try {
+    if (filterQuery) {
+      const parsedFilterQuery = JSON.parse(filterQuery);
+      if (
+        !parsedFilterQuery ||
+        ['string', 'number', 'boolean'].includes(typeof parsedFilterQuery) ||
+        Array.isArray(parsedFilterQuery)
+      ) {
+        throw new Error('expected value to be an object');
```

Comment:
I think this would be a `TypeError` to be more specific?

Reply:
I will create an issue because I would like to create a nice process for `TypeError`

Reply:
https://github.com/elastic/ingest-dev/issues/89